File: java/958. Check Completeness of a Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isCompleteTree(TreeNode root) {

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        boolean nullFound = false;

        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();

            if (node == null) {
                nullFound = true;
                continue;
            }

            if (nullFound) return false;

            queue.add(node.left);
            queue.add(node.right);
        }

        return true;
    }
}

File: python/958. Check Completeness of a Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:

        queue = deque([root])
        nullFound = False

        while len(queue) > 0:
            node = queue.popleft()

            if node == None:
                nullFound = True
                continue

            if nullFound:
                return False

            queue.append(node.left)
            queue.append(node.right)

        return True
    